R/FunctionReporter.R

```diff
@@ -643,9 +658,10 @@ FunctionReporter <- R6::R6Class(
     if (!is.list(x) && listable) {
         xList <- as.list(x)
 
+        if (length(xList) > 0){
```

Comment:
I found this part of the diff and the corresponding code block really difficult to read. Would you consider a rewrite like this?

```r
if (length(xList) > 0){
    # Check if expression x is from _$_
    if (identical(xList[[1]], quote(`$`))) {
        # Check if expression x is of form self$foo, private$foo, or super$foo
        # We want to keep those together because they could refer to the class'
        # methods. So expression is not listable
        if (xList[[2]] %in% c(quote(self), quote(private), quote(super))) {
            listable <- FALSE
        } else {
            # If expression lefthand side is not keyword, we still want to split
            # it up because left might be a function
            # but we want to get rid of right, because it's a symbol in left's namespace
            # and not a symbol that could be reliably matched to the package namespace
            x <- xList[1:2]
        }
    } else {
        # make empty list "non-listable" so recursion stops
        listable <- FALSE
    }
}
```

Reply:
sure. not a problem.

Reply:
@jameslamb revised.